File: FixMyCar/FixMyCar.UI/fixmycar_car_repair_shop/lib/src/models/order/order_search_object.dart
import 'package:json_annotation/json_annotation.dart';

part 'order_search_object.g.dart';

@JsonSerializable()
class OrderSearchObject {
  bool? discount;
  String? state;
  double? minTotalAmount;
  double? maxTotalAmount;
  DateTime? minOrderDate;
  DateTime? maxOrderDate;
  DateTime? minShippingDate;
  DateTime? maxShippingDate;

  OrderSearchObject.n({
    required this.minTotalAmount,
    required this.maxTotalAmount,
  });

  OrderSearchObject(
      this.discount,
      this.state,
      this.minTotalAmount,
      this.maxTotalAmount,
      this.minOrderDate,
      this.maxOrderDate,
      this.minShippingDate,
      this.maxShippingDate);

  factory OrderSearchObject.fromJson(Map<String, dynamic> json) =>
      _$OrderSearchObjectFromJson(json);

  Map<String?, dynamic> toJson() => _$OrderSearchObjectToJson(this);
}

File: FixMyCar/FixMyCar.Model/SearchObjects/OrderSearchObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FixMyCar.Model.SearchObjects
{
    public class OrderSearchObject : BaseSearchObject
    {
        public string? CarPartsShopName { get; set; }
        public string? CarRepairShopName { get; set; }
        public string? ClientName { get; set; }
        public string? Role { get; set; }
        public bool? Discount { get; set; }
        public string? State { get; set; }
        public double? MinTotalAmount { get; set; }
        public double? MaxTotalAmount { get; set; }
        public DateTime? MinOrderDate { get; set; }
        public DateTime? MaxOrderDate { get; set; }
        public DateTime? MinShippingDate { get; set; }
        public DateTime? MaxShippingDate { get; set; }
    }
}
